Refining functionality and code cleanup

CMeyer19 opened the pull request.

CMeyer19 committed:
Bumping version in order to create a PR

theadley reviewed (commented):
So far the only change is the version, so you can use this PR for more changes if you want me to review them later

CMeyer19 committed:
Code cleanup according to review comments

CMeyer19 committed:
Apply changes based on current change comments:
 - Adding fields to trackBy functions
 - Adding take(1) to access token header functionality in the api.
 - Using template literals for bearer token string.

CMeyer19 committed:
Updating comment to a TODO task

CMeyer19 committed:
Replacing ReplaySubject with normal Subject

CMeyer19 committed:
Adding HTTP interceptor for bearer token and other headers and resulting code cleanup

CMeyer19 committed:
Doing some project restructuring and adding some index files and some location aliases

CMeyer19 committed:
Resolving PR comments

CMeyer19 committed:
Resolving PR comments

CMeyer19 committed:
Resolving PR comments
